.github/workflows/python-bindings.yml

```diff
@@ -69,7 +75,7 @@ jobs:
       - name: Setup ccache
         uses: hendrikmuhs/ccache-action@v1.2
         with:
-          key: '${{matrix.os}}-pyfiction'
+          key: '${{matrix.os}}-${{matrix.toolset}}-pyfiction'
```

Comment:
I don't think you need the toolset key because there is only one build per operating system